django/db/migrations/autodetector.py

```diff
@@ -1304,13 +1304,30 @@ def create_altered_indexes(self):
 
     def generate_added_indexes(self):
         for (app_label, model_name), alt_indexes in self.altered_indexes.items():
+            dependencies = []
+            model_state = self.to_state.models[app_label, model_name]
+            fields = list(model_state.fields.values())
+
             for index in alt_indexes["added_indexes"]:
+
+                for field in fields:
+                    if field.is_relation:
+                        dependencies.extend(
+                            self._get_dependencies_for_foreign_key(
+                                app_label,
+                                model_name,
+                                field,
+                                self.to_state,
+                            ),
+                        )
+
```

Comment:
Any reason why this can't be moved out of the loop as well?

Reply:
Oops, you're right! I'll take the opportunity to refactor the section responsible for fetching the dependencies, and get rid of the duplicated code. Eagles eyes 🦅👀